cns/restserver/restserver.go

```diff
@@ -135,6 +137,21 @@ type containerstatus struct {
 	VfpUpdateComplete             bool // True when VFP programming is completed for the NC
 }
 
+type multiContainerStatus map[string]containerstatus
```

Comment:
+1. This does not need to be a type

Reply:
This variable is actually not needed as you are simply duplicating 'service.state.ContainerStatus'

Reply:
I somewhat disagree with the sentiment in this thread. This should be an `error` implementation if anything, but the problem as it currently stands is that `GetUnsuccessfulStatusErrors` can return `""` making it a poor `error` (since in its typical usage `err != nil`, but it is, in fact, nil).

So the idea of having some function determine whether or not there were errors is a good one, but it should actually produce an error or nil--not a string.

It should read something like this:

```go
err := anyFailed(service.state.ContainerStatus)
if err != nil {
       // create the response but use `err.Error()`
}
```

You could have an error type like this to better deal with this:

```go
type MultiContainerError map[string]v1alpha1.NCStatus

func (m *MultiContainerError) Error() string {
       out := bytes.NewBufferString("multiple NCs failed: ")
       for ncid, status := range m {
              fmt.Fprintf(out, "%s (%s) ")
       }
       return out.String()
}
```